controllers/config.go

```diff
@@ -102,6 +112,56 @@ func InitFromFlags(fs *flag.FlagSet, cfg *ManagerConfig) {
 
 	cfg.SecretControllerConfig = &secret.Config{}
 	secret.InitFromFlags(fs, cfg.SecretControllerConfig)
+
+	return nil
+}
+
+// initFeatureGates initializes feature gates from the provided CLI flag set.
+func initFeatureGates(fs *flag.FlagSet, mgrcfg *ManagerConfig) error {
```

Comment:
```suggestion
func initFeatureGates(fs *flag.FlagSet, mgrCfg *ManagerConfig) error {
```

Using `mgrCfg` or just `cfg` should be absolutely fine. `mgrcfg` does not follow correct nomenclature because `mgr` and `cfg` are abbreviations of two different words, so the first letter of the second word needs to be capitalized.